src/activity/activity.jsx

```diff
@@ -15,46 +18,111 @@ import Histogram from './components/histogram'
-    this.state = this.formatState(props.data)
+    this.state = this.formatState(props)
   }
 
-  formatState (data) {
+  formatState (props) {
     return {
-      granularity: GRANULARITIES.Daily,
-      facet: FACETS.Features
+      granularity:
+        (props.granularity && startCase(props.granularity)) ||
+        GRANULARITIES.Daily,
+      facet: (props.facet && startCase(props.facet)) || FACETS.Features,
+      data: props.data,
+      range: props.range || [new Date(), new Date()]
     }
   }
 
-  getUsers (data) {
-    return histogramUsers // .slice(0, 6)
-      .map(d => [d, _max(d)])
+  parseDate (d) {
+    const date = new Date(d)
+    const day = date.getDate() - 1
+    const month = date.getMonth()
+    const year = date.getFullYear()
+    return {
+      day,
+      month,
+      year,
+      len: monthLength(month, year)
+    }
+  }
+
+  formatData (data, getCount) {
+    const months = 12
+    const { range } = this.state
+    const fromSamp = new Date(range[0]).getTime()
+    const toSamp = new Date(range[1]).getTime()
+    const filteredValues = data
+      .sort((a, b) => a.day - b.day)
+      .filter(d => d && d.day >= fromSamp && d.day < toSamp)
+
+    const max = filteredValues.length
+      ? getCount(_maxBy(filteredValues, getCount))
+      : 0
+
+    return [
+      filteredValues.reduce((result, item) => {
+        const { day, month, year, len } = this.parseDate(item.day)
+        result[year] = !result[year] ? new Array(months) : result[year]
+        result[year][month] =
+          result[year][month] || new Array(len).fill(0, 0, len)
+
+        result[year][month].forEach((d, i) => {
+          if (i + 1 === day) {
+            result[year][month][i] = getCount(item)
+          }
+        })
+
+        return result
+      }, {}),
+      max
+    ]
   }
 
-  getFeatures (data) {
-    return histogramFeatures // .slice(0, 6)
-      .map(d => [d, _max(d)])
+  getFeatures () {
+    const getCount = d => d.count_features
+    return this.formatData(this.state.data.buildings.activity_count, getCount)
+  }
+
+  getUsers (data) {
+    const getCount = d => d.count_users
+    return this.formatData(this.state.data.buildings.activity_users, getCount)
   }
 
   // groups days by week and returns the average of each week
-  groupByWeek (data) {
-    return data.reduce((acc, [days, max]) => {
-      const weeks = chunk(days, 7).map(d => _mean(d))
-      acc.push([weeks, max])
-      return acc
-    }, [])
+  groupByWeek ([data, max]) {
+    return [
+      _reduce(
+        data,
+        (years, months, year) => {
+          years[year] =
+            years[year] ||
+            months.map(days => _chunk(days, 7).map(d => _mean(d)))
+
+          return years
+        },
+        {}
+      ),
+      max
+    ]
   }
 
   // groups days by month and returns the average of each Monthly
-  groupByMonth (data) {
-    return data.map(([d, max]) => [[_mean(d)], max])
+  groupByMonth ([data, max]) {
```

Comment:
`groupByMonth` looks very similar to `groupByWeek`. Is there any way we could factorize a bit?